server/controllers/activitypub/lib/post_activity.ts

```diff
@@ -28,19 +28,8 @@ export async function postActivity ({ actorName, inboxUri, bodyTo, activity }: {
     body,
   })
   postHeaders['content-type'] = 'application/activity+json'
-  try {
-    await requests_.post(inboxUri, {
-      headers: postHeaders,
-      body,
-      timeout,
-      parseJson: false,
-      retryOnceOnError: true,
-    })
-  } catch (err) {
-    err.context = err.context || {}
-    Object.assign(err.context, { inboxUri, activity })
-    logError(err, 'Posting activity to inbox failed')
-  }
+  await postActivityQueue.push({ inboxUri, postHeaders, body, activity })
```

Comment:
Aren't signed requests time sensitive? If the job takes too much time to be done (because a lot is going on or we later decide to `waitForCPUsLoadToBeBelow(x)`, I'm afraid some signed requests might fail. It seems that could easily be fixed by having the job work also do the part that calls `signRequest`, by passing all `postActivity` arguments as job payload data

Reply:
Right, I think it's 30 seconds for mastodon, which is quite some time for a job to be executed. But hey, indeed the fix is easy (done in 87d3624a5)